.github/workflows/dummy-agent-test.yml

```diff
@@ -1,6 +1,14 @@
 name: Run e2e test with dummy agent
 
-on: [push]
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
```

Comment:
this won't help unless people are using `opendevin`'s branches for development? IMO the best practice with GitHub is that people only work on their own forks, and don't submit a PR until they are almost ready.